core/domain/suggestion_services.py

```diff
@@ -1490,9 +1534,22 @@ def update_question_suggestion(
             'skill_difficulty': skill_difficulty
         })
     suggestion.pre_update_validate(new_change_obj)
-    suggestion.edited_by_reviewer = True
-    suggestion.change = new_change_obj
+    if suggestion.author_id != user_id:
+        suggestion.edited_by_reviewer = True
+
+    if suggestion.status == suggestion_models.STATUS_REJECTED:
+        create_suggestion(
```

Comment:
Same note, and why is the code duplicated? Perhaps refactor into a helper method to avoid duplication.